### PR TITLE
add support in parquet_encode for decimal,map,list types

### PR DESCRIPTION
**Pull request by ryanworl**

### CODE REVIEW

**Review by richardartoul — APPROVED** [review]

File: internal/impl/parquet/processor_encode.go [hunk 1]
```diff
@@ -73,8 +74,10 @@ func init() {
 func parquetSchemaConfig() *service.ConfigField {
 	return service.NewObjectListField("schema",
 		service.NewStringField("name").Description("The name of the column."),
-		service.NewStringEnumField("type", "BOOLEAN", "INT32", "INT64", "FLOAT", "DOUBLE", "BYTE_ARRAY", "UTF8").
+		service.NewStringEnumField("type", "BOOLEAN", "INT32", "INT64", "DECIMAL64", "DECIMAL32", "FLOAT", "DOUBLE", "BYTE_ARRAY", "UTF8", "MAP", "LIST").
 			Description("The type of the column, only applicable for leaf columns with no child fields. Some logical types can be specified here such as UTF8.").Optional(),
+		service.NewIntField("decimal_precision").Description("Precision to use for DECIMAL32/DECIMAL64 type").Default(0),
```
Comment (richardartoul) [inline internal/impl/parquet/processor_encode.go:79]: Can you also update the example above to include map/list in the schema? I assume they're probably a bit more confusing to configure cause the map has its own sub-schema right?

File: website/docs/components/processors/parquet_encode.md [hunk 2]
```diff
@@ -111,7 +111,23 @@ The type of the column, only applicable for leaf columns with no child fields. S
 
 
 Type: `string`  
-Options: `BOOLEAN`, `INT32`, `INT64`, `FLOAT`, `DOUBLE`, `BYTE_ARRAY`, `UTF8`.
+Options: `BOOLEAN`, `INT32`, `INT64`, `DECIMAL64`, `DECIMAL32`, `FLOAT`, `DOUBLE`, `BYTE_ARRAY`, `UTF8`, `MAP`, `LIST`.
```
Comment (richardartoul) [inline website/docs/components/processors/parquet_encode.md:114]: Maybe worth clarifying if map/list can (or cant?) be used recursively as well

File: internal/impl/parquet/processor_encode.go (hunk 1 above)
Comment (ryanworl) [reply internal/impl/parquet/processor_encode.go:88]: done

File: website/docs/components/processors/parquet_encode.md (hunk 2 above)
Comment (ryanworl) [reply website/docs/components/processors/parquet_encode.md:123]: done

**Pull request merged**